packages/dd-trace/src/debugger/devtools_client/queue.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+class JSONQueue {
+  constructor ({ size, timeout, onFlush }) {
+    this._maxSize = size
+    this._timeout = timeout
+    this._onFlush = onFlush
+    this._reset()
+  }
+
+  _reset () {
+    clearTimeout(this._timer)
+    this._timer = null
+    this._partialJson = null
+  }
+
+  _flush () {
+    const json = `${this._partialJson}]`
+    this._reset()
+    this._onFlush(json)
+  }
+
+  add (str, size = Buffer.byteLength(str)) {
```

Comment:
I think it better to pass just the json and remove the size param. 
If the user give wrong values it break the logic of the queue

Reply:
This is purely a performance optimization. In the snapshot code, the size is already calculated since we need to know if it's larger than 1MB. So to not calculate it twice we allow it to be passed in. It's not user generated input.